discovery/pkg/metrics/metrics.go

```diff
@@ -14,87 +14,150 @@
 			),
 			QueueSizeGauge: prometheus.NewGaugeVec(
 				prometheus.GaugeOpts{
 					Name: QueueSizeGauge,
 					Help: "Number of items in process queue",
 				},
-				[]string{"backendname", "clustertype"},
+				[]string{"backendname", "backendtype"},
 			),
 			DiscovererAPILatencyMSGauge: prometheus.NewGaugeVec(
 				prometheus.GaugeOpts{
 					Name: DiscovererAPILatencyMSGauge,
 					Help: "The milliseconds it takes for requests to return from a remote discoverer api",
 				},
-				[]string{"backendname", "clustertype", "path"},
+				[]string{"backendname", "backendtype", "path"},
 			),
 			DiscovererCycleDurationMSGauge: prometheus.NewGaugeVec(
 				prometheus.GaugeOpts{
 					Name: DiscovererCycleDurationMSGauge,
 					Help: "The milliseconds it takes for all objects to be synced from a remote discoverer api",
 				},
-				[]string{"backendname", "clustertype"},
+				[]string{"backendname", "backendtype"},
 			),
 			DiscovererErrorTotal: prometheus.NewCounterVec(
 				prometheus.CounterOpts{
 					Name: DiscovererErrorTotal,
 					Help: "Number of errors that have occurred in the Discoverer",
 				},
-				[]string{"backendname", "errortype"},
+				[]string{"backendname", "errortype", "backendtype"},
+			),
+			DiscovererUpstreamServicesGauge: prometheus.NewGaugeVec(
+				prometheus.GaugeOpts{
+					Name: DiscovererUpstreamServicesGauge,
+					Help: "Total number of services in the upstream backend cluster",
+				},
+				[]string{"backendname", "namespace", "backendtype"},
+			),
+			DiscovererReplicatedServicesGauge: prometheus.NewGaugeVec(
+				prometheus.GaugeOpts{
+					Name: DiscovererReplicatedServicesGauge,
+					Help: "Total number of services replicated from the backend cluster",
+				},
+				[]string{"backendname", "namespace", "backendtype"},
+			),
+			DiscovererInvalidServicesGauge: prometheus.NewGaugeVec(
+				prometheus.GaugeOpts{
+					Name: DiscovererInvalidServicesGauge,
+					Help: "Total number of invalid services that could not be replicated from the backend cluster",
+				},
+				[]string{"backendname", "namespace", "backendtype"},
+			),
+			DiscovererUpstreamEndpointsGauge: prometheus.NewGaugeVec(
+				prometheus.GaugeOpts{
+					Name: DiscovererUpstreamEndpointsGauge,
+					Help: "Total number of endpoints in the backend cluster",
+				},
+				[]string{"backendname", "namespace", "servicename", "backendtype"},
+			),
+			DiscovererReplicatedEndpointsGauge: prometheus.NewGaugeVec(
+				prometheus.GaugeOpts{
+					Name: DiscovererReplicatedEndpointsGauge,
+					Help: "Total number of endpoints replicated the backend cluster",
+				},
+				[]string{"backendname", "namespace", "servicename", "backendtype"},
+			),
+			DiscovererInvalidEndpointsGauge: prometheus.NewGaugeVec(
+				prometheus.GaugeOpts{
+					Name: DiscovererInvalidEndpointsGauge,
+					Help: "Total number of endpoints invalid endpoints that could not be replicaed from the backend cluster",
```

Comment:
typo: "endpoints invalid endpoints"
but not sure if we need this metric anyways.